### PR TITLE
Remove --rpc-only-in-reply flag from VTGate

### PR DESCRIPTION
**Pull request by aaijazi**
This isn't ready for review yet, I'm creating the PR to trigger Travis builds. I believe the Java client needs a good deal of work still.

### CODE REVIEW

File: go/vt/vtgate/gorpcvtgateservice/server.go [hunk 1]
```diff
@@ -376,18 +331,15 @@ func (vtg *VTGate) StreamExecuteKeyRanges2(ctx context.Context, request *proto.K
...
-		// problems where the client will get out of sync with the number of QueryResults sent.
-		// That's why this the error is only sent this way when the --rpc_errors_only_in_reply flag is set
-		// (signalling that all clients are able to handle new-style errors).
-		return sendReply(qr)
-	}
-	return vtgErr
+	// If there was an app error, send a QueryResult back with it.
+	qr := new(proto.QueryResult)
+	vtgate.AddVtGateError(vtgErr, &qr.Err)
+	// Sending back errors this way is not backwards compatible. If a (new) server sends an additional
```
Comment (alainjobart) [inline go/vt/vtgate/gorpcvtgateservice/server.go]: this comment needs to go I think

**Comment by alainjobart** [comment]
LGTM

File: go/vt/vtgate/gorpcvtgateservice/server.go (hunk 1 above)
Comment (aaijazi) [inline go/vt/vtgate/gorpcvtgateservice/server.go]: Done.

**Comment by aaijazi** [comment]
Submitting this change with passing tests. Note: this will break the gorpc Java client, as it will no longer handle errors correctly.

Next step is to remove the gorpc Java client entirely.

**Pull request merged**